analyses/transcriptomic-dimension-reduction/scripts/run-dimension-reduction.R

```diff
@@ -0,0 +1,221 @@
+    default = NULL,
+    help = "full path to expression data RDS file",
+    metavar = "integer"
+  ),
+  optparse::make_option(
+    c("-m", "--metadata"),
+    type = "character",
+    default = NULL,
+    help = "full path to metadata TSV file"
+  ),
+  optparse::make_option(
+    c("-o", "--output_directory"),
+    type = "character",
+    default = NULL,
+    help = "output directory"
+  ),
+  optparse::make_option(
+    c("-f", "--filename_lead"),
+    type = "character",
+    default = NULL,
+    help = "A character vector that will be used to name output files"
+  ),
+  optparse::make_option(
+    c("-s", "--seed"),
+    type = "integer",
+    default = 2019,
+    help = "seed integer",
+    metavar = "integer"
+  ),
+  optparse::make_option(
+    c("-p", "--perplexity"),
+    type = "integer",
+    default = 10,
+    help = "t-SNE perplexity parameter integer",
+    metavar = "integer"
+  ),
+  optparse::make_option(
+    c("-n", "--neighbors"),
+    type = "integer",
+    default = 15,
+    help = "UMAP neighbors parameter integer",
+    metavar = "integer"
+  ),
+  optparse::make_option(
+    c("-l", "--low_count_threshold"),
+    type = "integer",
+    default = 100,
+    help = "Number of total counts required to retain a gene",
+    metavar = "integer"
+  ),
+  optparse::make_option(
+    "--skip_tsne",
+    action = "store_true",
+    default = FALSE,
+    help = "If used, t-SNE will be skipped"
+  )
+)
+
+# Read the arguments passed
+opt_parser <- optparse::OptionParser(option_list = option_list)
+opt <- optparse::parse_args(opt_parser)
+
+# RDS file that contains the expression data
+expression_file <- opt$expression
+
+# TSV file that contains metadata
+metadata_file <- opt$metadata
+
+# Create specified output directory if it does not yet exist
+output_directory <- opt$output_directory
+if (!dir.exists(output_directory)) {
+  dir.create(output_directory, recursive = TRUE)
+}
+
+# Character string that will be used to name files
+filename_lead <- opt$filename_lead
+
+# Define seed as the parmeter passed via the command line
+seed <- opt$seed
+
+# Define t-SNE perplexity parameter as the parameter passed via command line
+perplexity_parameter <- opt$perplexity
+
+# Define UMAP neighbors parameter as the parameter passed via command line
+neighbors_parameter <- opt$neighbors
+
+# Threshold used to filter out genes with low counts
+gene_count_threshold <- opt$low_count_threshold
+
+#### Read in data --------------------------------------------------------------
+
+# Read in metadata
+metadata_df <- data.frame(readr::read_tsv(metadata_file))
+
+# Read in expression data
+expression_data <- readr::read_rds(expression_file) %>%
+  as.data.frame()
+
+# the first column will be either transcript or column ids
+feature_identifier <- colnames(expression_data)[1]
+
+# drop any columns that contain other identifers
+expression_data <- expression_data %>%
+  dplyr::select(!!rlang::sym(feature_identifier), dplyr::starts_with("BS_")) %>%
```

Comment:
This drops the `gene_id` column in the kallisto dataset because `transcript_id` is the first column (what is captured by `feature_identifier` above). This line retains the first column (name is stored as `feature_identifier`) and the samples (`starts_with("BS_")` for the biospecimen IDs). In what you had earlier, you were dropping duplicate `gene_id` values, probably because you couldn't set them as `rownames`, and the matrix was losing a lot of rows as a result. (I didn't dig into how many rows were removed too much, it was something I happened to notice.)

Reply:
Understood.